File: entities/dlq.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type VerificationDeadJob struct {
	ID            uuid.UUID `gorm:"primaryKey" json:"id"`
	JobID         uuid.UUID `gorm:"not null" json:"job_id"`
	DoctorID      uuid.UUID `gorm:"not null;index" json:"doctor_id"`
	FailureReason string    `gorm:"type:text;not null" json:"failure_reason"`
	RetryCount    int       `gorm:"default:0;not null" json:"retry_count"`
	PayloadJSON   *string   `gorm:"type:text" json:"payload_json,omitempty"`
	FailedAt      time.Time `gorm:"autoCreateTime" json:"failed_at"`
}

func (d *VerificationDeadJob) BeforeCreate(tx *gorm.DB) (err error) {
	if d.ID == uuid.Nil {
		d.ID = uuid.New()
	}
	return nil
}

File: python_backend/entities/dlq.py
import uuid
from sqlalchemy import Column, String, Integer, DateTime, Text, ForeignKey
from sqlalchemy.sql import func
from database import Base

class VerificationDeadJob(Base):
    __tablename__ = "verification_dead_jobs"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    job_id = Column(nullable=False)
    doctor_id = Column(String(36), ForeignKey("doctors.id", ondelete="CASCADE"), nullable=False, index=True)
    failure_reason = Column(Text, nullable=False)
    retry_count = Column(Integer, default=0, nullable=False)
    payload_json = Column(Text, nullable=True)
    failed_at = Column(DateTime(timezone=True), server_default=func.now())
